xmake/rules/c++/modules/modules_support/builder.lua

```diff
@@ -207,10 +198,10 @@ function build_modules_for_batchjobs(target, batchjobs, sourcebatch, modules, op
 
     local modulesjobs = {}
     _build_modules(target, sourcebatch, modules, table.join(opt, {
-       build_module = function(deps, build, module, name, provide, objectfile, cppfile, fileconfig)
+       build_module = function(deps, module, name, provide, objectfile, cppfile, fileconfig)
         local job_name = name and target:name() .. name or cppfile
 
-        modulesjobs[job_name] = _builder(target).make_module_buildjobs(target, batchjobs, job_name, deps, {build = build, module = module, objectfile = objectfile, cppfile = cppfile})
+        modulesjobs[job_name] = _builder(target).make_module_buildjobs(target, batchjobs, job_name, deps, _should_build, _mark_build, {module = module, objectfile = objectfile, cppfile = cppfile})
```

Comment:
if other submodule need to call _should_build and _mark_build, we should mark them as public and call `should_build` directly